sql/expression/in.go

```diff
@@ -260,6 +267,39 @@ func (hit *HashInTuple) Eval(ctx *sql.Context, row sql.Row) (interface{}, error)
 	return true, nil
 }
 
+// convertOrTruncate converts the value |i| to type |t| and returns the converted value; if the value does not convert
+// cleanly and the type is automatically coerced (i.e. string and numeric types), then a warning is logged and the
+// value is truncated to the Zero value for type |t|. If the value does not convert and the type is not automatically
+// coerced, then an error is returned.
+func convertOrTruncate(ctx *sql.Context, i interface{}, t sql.Type) (interface{}, error) {
+	converted, err := t.Convert(i)
+	if err == nil {
+		return converted, nil
+	}
+
+	// If a value can't be converted to an enum or set type, truncate it to a value that is guaranteed
+	// to not match any enum value.
+	if sql.IsEnum(t) || sql.IsSet(t) {
```

Comment:
maybe a nit, but it looks like you did embed the enum case, and we can't return an error for a failed enum cast. Can the enum case not be reached? Are there other errors we are masking here that we shouldn't be?

Reply:
Enums got an update to their `Compare` method (not `Convert`), so `Compare` is more permissive about the values passed in and won't error out on comparing invalid enum values any more. `Convert` will still throw errors though, which we do want in other places so that we don't always silently swallow a conversion error.